frontend/src/utils/context.tsx

```diff
@@ -18,3 +40,73 @@ export default function useFocusedUser(): AppContextType["focusedUser"] {
   const context = React.useContext(AppContext)
   return context.focusedUser
 }
+
+type UseUserReturnType = {
+  user: Nullable<User>
+  logged_in: Nullable<boolean>
+  register: (authInfo: AuthInformation, callBack: CallBack) => void
+  login: (authInfo: AuthInformation, callBack: CallBack) => void
+  logout: (CallBack: CallBack) => void
```

Comment:
Instead of passing a callback to these functions, it would be nicer to return the promise associated with the request. As it stands right now, there is no error handling in these requests and no way to do it from the caller.

Reply:
Interesting idea, I'll explore that

Reply:
As far as I can tell, the two options for a return type are `Promise<void>` and `Promise<void | HttpResponse<Response>>>` Neither are great, but I'll settle for the former since we're not sending data back

Reply:
Those functions don't return anything for the final `then` so `Promise<void>` is fine.